src/main/java/org/broadinstitute/hellbender/tools/walkers/variantutils/FamilyLikelihoods.java

```diff
@@ -0,0 +1,455 @@
+        final double[] childLog10Posteriors = getPosteriors(FamilyMember.CHILD);
+
+        final double[] motherPosteriors = GeneralUtils.normalizeFromLog10(motherLog10Posteriors);
+        final double[] fatherPosteriors = GeneralUtils.normalizeFromLog10(fatherLog10Posteriors);
+        final double[] childPosteriors = GeneralUtils.normalizeFromLog10(childLog10Posteriors);
+
+
+        double jointPosteriorProbability =  -1;
+        //jointTrioLikelihood is combined likelihoods (before prior) of best configuration after applying prior
+        double jointTrioLikelihood = -1;
+        if(childIsCalled && motherIsCalled && fatherIsCalled) {
+            jointTrioLikelihood = motherLikelihoods[MathUtils.maxElementIndex(motherPosteriors)]*fatherLikelihoods[MathUtils.maxElementIndex(fatherPosteriors)]*childLikelihoods[MathUtils.maxElementIndex(childPosteriors)];
+            jointPosteriorProbability = MathUtils.arrayMax(motherPosteriors)*MathUtils.arrayMax(fatherPosteriors)*MathUtils.arrayMax(childPosteriors);
+        }
+
+        updatedGenotypes.add(getUpdatedGenotype(vc, motherGenotype, jointTrioLikelihood, jointPosteriorProbability, motherLog10Posteriors));
+        updatedGenotypes.add(getUpdatedGenotype(vc, fatherGenotype, jointTrioLikelihood, jointPosteriorProbability, fatherLog10Posteriors));
+        updatedGenotypes.add(getUpdatedGenotype(vc, childGenotype, jointTrioLikelihood, jointPosteriorProbability, childLog10Posteriors));
+    }
+
+    private Genotype getUpdatedGenotype(final VariantContext vc, final Genotype genotype, final double jointLikelihood, final double jointPosteriorProb, final double[] log10Posteriors){
+        //Don't update null, missing or unavailable genotypes
+        if(genotype == null || !hasCalledGT(genotype.getType())) {
+            return genotype;
+        }
+
+        int phredScaledJL = -1;
+        int phredScaledJP = -1;
+        if(jointLikelihood != NO_JOINT_VALUE){
+            final double dphredScaledJL = QualityUtils.phredScaleLog10ErrorRate(Math.log10(1-jointLikelihood));
+            phredScaledJL = dphredScaledJL < Byte.MAX_VALUE ? (byte)dphredScaledJL : Byte.MAX_VALUE;
+        }
+        if(jointPosteriorProb != NO_JOINT_VALUE){
+            final double dphredScaledJP = QualityUtils.phredScaleLog10ErrorRate(Math.log10(1-jointPosteriorProb));
+            phredScaledJP = dphredScaledJP < Byte.MAX_VALUE ? (byte)dphredScaledJP : Byte.MAX_VALUE;
+        }
+
+        //Add the joint trio calculations
+        final Map<String, Object> genotypeAttributes = new HashMap<>();
+        genotypeAttributes.putAll(genotype.getExtendedAttributes());
+        genotypeAttributes.put(GATKVCFConstants.JOINT_LIKELIHOOD_TAG_NAME, phredScaledJL);
+        genotypeAttributes.put(GATKVCFConstants.JOINT_POSTERIOR_TAG_NAME, phredScaledJP);
+
+        final GenotypeBuilder builder = new GenotypeBuilder(genotype);
+
+        //final double[] log10Posteriors = MathUtils.toLog10(normalizedPosteriors);
```

Comment:
This should probably be deleted or a comment explaining why it's here should be added.